### PR TITLE
Feat/implement sdl2 lib

### LINKED ISSUE #22: Implement SDL2 lib
# :white_check_mark: What's needed
Implement the SDL2lib

## 📋 Subtasks
- [x] rename SDL2_lib by SDL2lib
- [x] closeWindow
- [x] clearWindow
- [x] displayEntities
- [x] displayText
- [x] playSound

## :file_folder: Files required
src/Graphical/SDL2/sdl2.hpp
src/Graphical/SDL2/sdl2.cpp

### PR DESCRIPTION
**Pull request by mathieurobert1**
Implement the SDL2 lib .so
I create all the methods define in the IGraphic for the SDL2lib class.

More details : #22

### CODE REVIEW

File: src/Graphical/SDL2/sdl2.cpp [hunk 1]
```diff
@@ -17,62 +19,118 @@ SDL2_lib::SDL2_lib()
...
-bool SDL2_lib::isWindowOpen() const
+bool SDL2lib::isWindowOpen() const
 {
     return _isOpen;
 }
 
-void SDL2_lib::closeWindow()
+void SDL2lib::closeWindow()
 {
+    _isOpen = false;
```
Comment (Thomaltarix) [inline src/Graphical/SDL2/sdl2.cpp:50]: Does i close the window ? :thinking:

File: src/Graphical/SDL2/sdl2.cpp [hunk 2]
```diff
@@ -17,62 +19,118 @@ SDL2_lib::SDL2_lib()
...
+void SDL2lib::displayWindow()
 {
     SDL_RenderPresent(_Renderer);
 }
 
-void SDL2_lib::displayEntities(std::vector<std::shared_ptr<IEntity>> entities)
+void SDL2lib::displayEntities(std::vector<std::shared_ptr<IEntity>> entities)
+{
+    for (auto &entity : entities) {
+        std::unique_ptr<SDL_Texture, decltype(&SDL_DestroyTexture)> texture = getEntityTexture(entity.get()->getPath());
```
Comment (Thomaltarix) [inline src/Graphical/SDL2/sdl2.cpp:87]: Is `.get()` necessary ?

Comment (Thomaltarix) [reply src/Graphical/SDL2/sdl2.cpp:87]: (I'll not ping all those, but they're many others)

**Review by Thomaltarix — APPROVED** [review]
Fine for me :ok_hand:

File: src/Graphical/SDL2/sdl2.cpp (hunk 2 above)
Comment (mathieurobert1) [reply src/Graphical/SDL2/sdl2.cpp:87]: I don't know, I thought it was necessary.

**Pull request merged**